linera-execution/tests/test_execution.rs

```diff
@@ -1621,3 +1621,52 @@ async fn test_close_chain() {
     .unwrap();
     assert!(view.system.closed.get());
 }
+
+/// Tests the system API call `read_blob``.
+#[tokio::test]
+async fn test_failing_read_blob() {
+    let committee = Committee::make_simple(vec![PublicKey::test_key(0).into()]);
+    let committees = BTreeMap::from([(Epoch::ZERO, committee)]);
+    let ownership = ChainOwnership::single(PublicKey::test_key(1));
+    let state = SystemExecutionState {
+        committees: committees.clone(),
+        ownership: ownership.clone(),
+        balance: Amount::from_tokens(5),
+        ..SystemExecutionState::new(Epoch::ZERO, ChainDescription::Root(0), ChainId::root(0))
+    };
+    let mut view = state.into_view().await;
+    let mut applications = register_mock_applications(&mut view, 1).await.unwrap();
+    let (application_id, application) = applications.next().unwrap();
+
+    // The application is not authorized to close the chain.
+    let context = make_operation_context();
+    let blob = HashedBlob::test_blob("test");
+    let blob_id = blob.id();
+    application.expect_call(ExpectedCall::execute_operation(
+        move |runtime, _context, _operation| {
+            assert_matches!(
+                runtime.read_blob(&blob_id),
+                Err(ExecutionError::MissingRuntimeResponse)
```

Comment:
I'm not sure why we get this `MissingRuntimeResponse` here. Is it because we're in a testing environment? Is this fine for testing or is there something I can do to get a result closer to production here?